include/aws/http/http2_stream_manager.h

```diff
@@ -105,6 +105,26 @@ struct aws_http2_stream_manager_options {
     void *shutdown_complete_user_data;
     aws_http2_stream_manager_shutdown_complete_fn *shutdown_complete_callback;
 
+    /**
+     * Optional.
+     * When set, connection will be closed if 5xx response received from server.
+     */
+    bool close_connection_on_server_error;
```

Comment:
Is there anything else that might make the connection undesirable? If we had the API for marking a connection undesirable, then the S3 client could have its own specific logic, instead of baking the logic in at this low level

Reply:
The stream manager user doesn't have control of the stream, so that I don't think high level user will be able to do it by themselves.

Reply:
we'll cross that bridge if we come to it